common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -318,6 +317,8 @@ def load_item(self, location, for_parent=None):  # pylint: disable=method-hidden
                         ).replace(tzinfo=UTC)
                     module._edit_info['published_by'] = raw_metadata.get('published_by')
 
+                module._field_data = OverrideModulestoreFieldData.wrap(module)  # pylint: disable=protected-access
```

Comment:
@cpennington this used to be close to the former of your two suggestions:
```python
module._field_data = OverrideModulestoreFieldData.wrap(module, module._field_data)
```
I changed it because it seems silly to require `module._field_data` as an argument when that attribute is present on the `module`. The `module` is required because it's used to determine which field overrides to apply, and also to cache the result of that calculation (using the `location` attribute).